src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1292,4 +1299,1372 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+                }
+            }
+            if (!thisHP.companionHeatPumpCoil) {
+                ShowSevereError(state, "Could not find matching companion heat pump coil.");
+                ShowContinueError(state, format("Base coil: {}", thisCoilName));
+                ShowContinueError(state, format("Looking for companion coil named: {}", targetCompanionName));
+                ShowFatalError(state, "Simulation aborts due to previous severe error");
+            }
+        }
+    }
+}
+
+void EIRFuelFiredHeatPump::processInputForEIRPLHP(EnergyPlusData &state)
+{
+    struct ClassType
+    {
+        DataPlant::PlantEquipmentType thisType;
+        std::string nodesType;
+        std::function<Real64(Real64, Real64)> calcLoadOutletTemp;
+        std::function<Real64(Real64, Real64)> calcQsource;
+        std::function<Real64(Real64, Real64)> calcSourceOutletTemp;
+
+        ClassType(DataPlant::PlantEquipmentType _thisType,
+                  std::string _nodesType,
+                  std::function<Real64(Real64, Real64)> _tLoadOutFunc,
+                  std::function<Real64(Real64, Real64)> _qSrcFunc,
+                  std::function<Real64(Real64, Real64)> _tSrcOutFunc)
+            : thisType(_thisType), nodesType(std::move(_nodesType)), calcLoadOutletTemp(std::move(_tLoadOutFunc)), calcQsource(std::move(_qSrcFunc)),
+              calcSourceOutletTemp(std::move(_tSrcOutFunc))
+        {
+        }
+    };
+    std::vector<ClassType> classesToInput = {
+        ClassType{DataPlant::PlantEquipmentType::HeatPumpFuelFiredCooling,
+                  "Chilled Water Nodes",
+                  EIRPlantLoopHeatPumps::EIRFuelFiredHeatPump::subtract,
+                  EIRPlantLoopHeatPumps::EIRFuelFiredHeatPump::add,
+                  EIRPlantLoopHeatPumps::EIRFuelFiredHeatPump::add},
+        ClassType{DataPlant::PlantEquipmentType::HeatPumpFuelFiredHeating,
+                  "Hot Water Nodes",
+                  EIRPlantLoopHeatPumps::EIRFuelFiredHeatPump::add,
+                  EIRPlantLoopHeatPumps::EIRFuelFiredHeatPump::subtract,
+                  EIRPlantLoopHeatPumps::EIRFuelFiredHeatPump::subtract},
+    };
+
+    bool errorsFound = false;
+    std::string &cCurrentModuleObject = state.dataIPShortCut->cCurrentModuleObject;
+    for (auto &classToInput : classesToInput) {
+        cCurrentModuleObject = DataPlant::PlantEquipTypeNames[static_cast<int>(classToInput.thisType)];
+
+        DataLoopNode::ConnectionObjectType objType = static_cast<DataLoopNode::ConnectionObjectType>(
+            getEnumerationValue(BranchNodeConnections::ConnectionObjectTypeNamesUC, UtilityRoutines::MakeUPPERCase(cCurrentModuleObject)));
+
+        if (state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, cCurrentModuleObject) > 0) {
+            auto const instances = state.dataInputProcessing->inputProcessor->epJSON.find(cCurrentModuleObject);
+            if (instances == state.dataInputProcessing->inputProcessor->epJSON.end()) {
```

Comment:
Couldn't you just use continue here? and therefore not need to check getNumObjectsFound.

    if (instances == state.dataInputProcessing->inputProcessor->epJSON.end()) continue;

Reply:
Thanks @rraustad ! This seems to be a good idea. I tested it and it seem to be working fine. Obviously, here the `Severe Error` here should not be thrown and then also the `errorsFound` flag should not be set any more:
```
            // ShowSevereError(state,                                                                     // LCOV_EXCL_LINE
            //                 "EIR PLFFHP: Somehow getNumObjectsFound was > 0 but epJSON.find found 0"); // LCOV_EXCL_LINE
            // errorsFound = true;                                                                        // LCOV_EXCL_LINE
            continue;
```